### PR TITLE
ADEN-6200 Refactor F2 video ads tests

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    AdsJWPlayerObject jwPlayerObject = new AdsJWPlayerObject(driver);
+
+    jwPlayerObject.waitForAdPlaying();
+    pageObject.scrollTo(driver.findElement(AdsJWPlayerObject.PLAYER_SELECTOR));
+    jwPlayerObject.clickVolumeButton();
+    jwPlayerObject.allowToPlayVideoForSomeTime(Duration.ofSeconds(3));
+    Assertion.assertTrue(jwPlayerObject.wasSoundHeard());
+  }
+
+  private void verifyFeaturedVideoWithoutSound(String pageUrl) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java]: MAJOR Remove this unused private "verifyFeaturedVideoWithoutSound" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      groups = {"AdsFeaturedVideoSoundMercury"}
+  )
+  public void adsFeaturedVideoWithoutSoundMercury() {
+    verifyFeaturedVideoWithoutSound(AdsDataProvider.PAGE_FV_JWPLAYER_AND_SOUND.getUrl());
+  }
+
+  private void verifyFeaturedVideoWithSound(String pageUrl) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java]: MAJOR Remove this unused private "verifyFeaturedVideoWithSound" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/ad22ab242d9e5ea30607b50508e5bf7a3a2f75e4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/ad22ab242d9e5ea30607b50508e5bf7a3a2f75e4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/ad22ab242d9e5ea30607b50508e5bf7a3a2f75e4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/ad22ab242d9e5ea30607b50508e5bf7a3a2f75e4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/ac80f9870ef21af835f638b216a1e29c2dd42f19/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/ac80f9870ef21af835f638b216a1e29c2dd42f19/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/ac80f9870ef21af835f638b216a1e29c2dd42f19/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/ac80f9870ef21af835f638b216a1e29c2dd42f19/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 3]
```diff
@@ -154,9 +154,15 @@ public String getUrl() {
   /**
    * Checks if container containing all banner notification is present, it may be present, but may have 0px height
    * when no notification is displayed
+   *
+   * TODO: Move this logic to better place (WebElement?)
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: INFO Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/51e864ed32d5f63aeca4886235e0c87f7452bff1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/51e864ed32d5f63aeca4886235e0c87f7452bff1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/51e864ed32d5f63aeca4886235e0c87f7452bff1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/51e864ed32d5f63aeca4886235e0c87f7452bff1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/d578f2d3e2850b0eaa40bde4503b3f94b3911c36/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/d578f2d3e2850b0eaa40bde4503b3f94b3911c36/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/d578f2d3e2850b0eaa40bde4503b3f94b3911c36/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/d578f2d3e2850b0eaa40bde4503b3f94b3911c36/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/63ab49f33c07ba8820bd13c40591f2194b0485f4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/63ab49f33c07ba8820bd13c40591f2194b0485f4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/63ab49f33c07ba8820bd13c40591f2194b0485f4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/63ab49f33c07ba8820bd13c40591f2194b0485f4/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java [hunk 4]
```diff
@@ -50,21 +51,24 @@ public void adsFeaturedVideoWithoutSoundMercury() {
...
 
     jwPlayerObject.waitForAdPlaying();
-    jwPlayerObject.scrollToPlayer();
+    pageObject.scrollTo(driver.findElement(AdsJWPlayerObject.PLAYER_SELECTOR));
     jwPlayerObject.clickVolumeButton();
     jwPlayerObject.allowToPlayVideoForSomeTime(Duration.ofSeconds(3));
     Assertion.assertTrue(jwPlayerObject.wasSoundHeard());
   }
 
   private void verifyFeaturedVideoWithoutSound(String pageUrl) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java]: MAJOR Remove this unused private "verifyFeaturedVideoWithoutSound" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java [hunk 5]
```diff
@@ -50,21 +51,24 @@ public void adsFeaturedVideoWithoutSoundMercury() {
   }
 
   private void verifyFeaturedVideoWithSound(String pageUrl) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFeaturedVideoSound.java]: MAJOR Remove this unused private "verifyFeaturedVideoWithSound" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/a1bf613cbe64c6779331167cffd010e85cce4f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/a1bf613cbe64c6779331167cffd010e85cce4f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/a1bf613cbe64c6779331167cffd010e85cce4f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/a1bf613cbe64c6779331167cffd010e85cce4f84/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.dataprovider.ads.FandomAdsDataProvider;
+import com.wikia.webdriver.common.templates.fandom.AdsFandomTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsFandomObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsJWPlayerObject;
+import org.testng.annotations.Test;
+
+public class TestAdsFandomArticleVideo extends AdsFandomTestTemplate {
+  @Test(
+      groups = {"AdsArticleVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java]: Please add here `Desktop`

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    jwPlayerObject.verifyPlayerOnPage();
+    pageObject.scrollTo(AdsJWPlayerObject.VIDEO_PLAYER_SELECTOR);
+    jwPlayerObject.clickOnPlayer();
+    jwPlayerObject.verifyPreroll();
+  }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsArticleVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java]: ... and here

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    jwPlayerObject.clickOnPlayer();
+    pageObject.wait.forSuccessfulResponse(networkTrafficInterceptor, FandomAdsDataProvider.MOAT_VIDEO_TRACKING_URL);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      groups = {"AdsArticleVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java]: ...and `Mobile`

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    adsArticleVideoAdsDesktop();
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsArticleVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomArticleVideo.java]: ...aaaaaamnd here

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.dataprovider.ads.FandomAdsDataProvider;
+import com.wikia.webdriver.common.templates.fandom.AdsFandomTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsFandomObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsJWPlayerObject;
+import org.testng.annotations.Test;
+
+public class TestAdsFandomFeaturedVideo extends AdsFandomTestTemplate {
+  @Test(
+      groups = {"AdsFeaturedVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java]: ...and again :(

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+    loadPage(testedPage);
+    AdsJWPlayerObject jwPlayerObject = new AdsJWPlayerObject(driver);
+
+    jwPlayerObject.verifyPreroll();
+  }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsFeaturedVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java]: Same here :)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    jwPlayerObject.verifyPlayerOnPage();
+    pageObject.wait.forSuccessfulResponse(networkTrafficInterceptor, FandomAdsDataProvider.MOAT_VIDEO_TRACKING_URL);
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @Test(
+      groups = {"AdsFeaturedVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java]: ...and don't forget about this one

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    adsFeaturedVideoAdsDesktop();
+  }
+
+  @InBrowser(
+      browser = Browser.CHROME,
+      emulator = Emulator.GOOGLE_NEXUS_5
+  )
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsFeaturedVideoF2"}
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsFandomFeaturedVideo.java]: ... and this

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsJWPlayerObject.java [hunk 14]
```diff
@@ -1,66 +1,102 @@
...
-  private static final By PLAYER_SELECTOR = By.cssSelector(FEATURED_VIDEO_PLAYER_SELECTOR);
-  private static final By VOLUME_BUTTON_SELECTOR = By.cssSelector(".jwplayer div.jw-icon.jw-icon-volume");
+  public static final By AD_SELECTOR = By.cssSelector(VIDEO_AD_SELECTOR);
+  public static final By MOVIE_SELECTOR = By.cssSelector(VIDEO_MOVIE_SELECTOR);
+  public static final By PLAYER_SELECTOR = By.cssSelector(VIDEO_PLAYER_SELECTOR);
+  public static final By VOLUME_BUTTON_SELECTOR = By.cssSelector(".jwplayer div.jw-icon.jw-icon-volume");
 
-  public AdsJWPlayerObject(WebDriver driver, String page) {
-    super(driver, page);
+  public AdsJWPlayerObject(WikiaWebDriver driver) {
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsJWPlayerObject.java]: <3

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/c4726b336ced02dc66a2fe2bd4e66858c80c40f0/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/c4726b336ced02dc66a2fe2bd4e66858c80c40f0/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/c4726b336ced02dc66a2fe2bd4e66858c80c40f0/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/c4726b336ced02dc66a2fe2bd4e66858c80c40f0/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/871130052d6624ff816ed9f778fd77491238cfba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/871130052d6624ff816ed9f778fd77491238cfba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/871130052d6624ff816ed9f778fd77491238cfba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/871130052d6624ff816ed9f778fd77491238cfba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![INFO][INFO] [WikiBasePageObject.java#L161](https://github.com/Wikia/selenium-tests/blob/871130052d6624ff816ed9f778fd77491238cfba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L161): Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**